src/app/storage/duck/operations.ts

```diff
@@ -159,8 +159,8 @@ const fetchStorage = () => {
       const res = await client.list(resource);
       //temporary for ui work
       const migStorages = res.data.items;
-      const refs = await Promise.all(fetchMigStorageRefs(client, migMeta, migStorages));
-      const groupedStorages = groupStorages(migStorages, refs);
+      const secretRefs = await fetchMigStorageSecretRefs(client, migMeta, migStorages);
```

Comment:
+1 with these renames, I was going to suggest we should be more specific with these names as secrets.

This is breaking the way that the flow of the function works. I think we want to use `Promise.allSettled` for this [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled), [Proposal compliant shim](https://www.npmjs.com/package/promise.allsettled), ex:

`const secretRefResults = await Promise.allSettled(fetchMigStorageSecretRefs(client, migMeta, migStorages));`

Then you can determine the set of successful, fully loaded `MigStorages` from the ones that failed.

Following that, you can then dispatch an update and potentially handle failed `MigStorages`, which we should mark as not ready.